crates/c-api/src/lib.rs

```diff
@@ -722,12 +732,16 @@ pub unsafe extern "C" fn wasm_instance_new(
         }
         return ptr::null_mut();
     }
-    match Instance::new(module, &externs) {
+    record_instantiate(Instance::new(module, &externs), result)
+}
+
+unsafe fn record_instantiate(
```

Comment:
Extreme nit: `record_instantiate` implies a record is kept some somewhere to me. Perhaps `handle_instantiation` since it takes `Result<Instance>`?